Authenticate the committed Gradle wrapper JAR against Gradle's published checksum (#747)
aunger/gallery-button-for-pixel-camera: gradle/README.md

`distributionSha256Sum` pins the Gradle 8.9 distribution (`gradle-8.9-bin.zip`) to its published SHA-256.
The Gradle wrapper verifies the downloaded distribution against this checksum before running it, so a substituted or corrupted distribution fails loudly.

The wrapper JAR (`wrapper/gradle-wrapper.jar`) is committed and reviewed directly; it is read before `verification-metadata.xml` is consulted, so it cannot be covered by that file.

> Instead, `scripts/test_verification_metadata.sh` check (g) pins the JAR's SHA-256 to the checksum Gradle officially publishes for it (issue #744), so a substituted or corrupted wrapper JAR fails the `shell-tests` CI job rather than being trusted silently.

Author: ```suggestion
Instead, `scripts/test_verification_metadata.sh` check (g) authenticates the JAR's SHA-256 against its record of the checksum Gradle officially publishes for it (issue #744), so a substituted or corrupted wrapper JAR fails the `shell-tests` CI job rather than being trusted silently.
```

I initially understood "pins" to mean *sets* the pinned value, not  *checks* the pinned value. This script can't set it though.

My suggestion probably also isn't the best way to say this; feel free to tweak for clarity.

Author: 🤖 Author

Good catch, that ambiguity is real: "pins" reads as though the script *sets* the pinned value, when all it does is *check* against it. Fixed in `34d17f5`. I went with "verifies" rather than "authenticates" so the verb itself signals a check, and "a pinned copy of the checksum Gradle officially publishes" to locate the stored value without implying the script writes it:

> Instead, `scripts/test_verification_metadata.sh` check (g) verifies the JAR's SHA-256 against a pinned copy of the checksum Gradle officially publishes for it (issue #744), so a substituted or corrupted wrapper JAR fails the `shell-tests` CI job rather than being trusted silently.

This stays consistent with the maintenance line just below, which is where the pinned copy actually gets set ("update its pinned SHA-256 in `scripts/test_verification_metadata.sh`"). Suite still passes 8/8 locally.
